phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKeyIT.java

```diff
@@ -498,18 +508,29 @@ public void run() {
         exec.shutdownNow();
 
         int finalResult = nThreads * nCommits * nIncrementsPerCommit;
-        //assertEquals(finalResult,resultHolder[0]);
-        ResultSet rs = conn.createStatement().executeQuery("SELECT * FROM " + tableName + " WHERE counter1 >= 0");
+        boolean isIndexCreated = this.indexDDL != null && this.indexDDL.length() > 0;
+
+        ResultSet rs;
+        String selectSql = "SELECT * FROM " + tableName + " WHERE counter1 >= 0";
+        if (isIndexCreated) {
+            rs = conn.createStatement().executeQuery("EXPLAIN " + selectSql);
+            String actualExplainPlan = QueryUtil.getExplainPlan(rs);
+            IndexToolIT.assertExplainPlan(this.indexDDL.contains("local"), actualExplainPlan,
+                tableName, tableName + "_IDX");
+        }
+        rs = conn.createStatement().executeQuery(selectSql);
         assertTrue(rs.next());
         assertEquals("a",rs.getString(1));
         assertEquals(finalResult,rs.getInt(2));
         assertFalse(rs.next());
 
-        rs = conn.createStatement().executeQuery("SELECT /*+ NO_INDEX */ * FROM " + tableName + " WHERE counter1 >= 0");
-        assertTrue(rs.next());
-        assertEquals("a",rs.getString(1));
-        assertEquals(finalResult,rs.getInt(2));
-        assertFalse(rs.next());
+        if (isIndexCreated) {
+            rs = conn.createStatement().executeQuery("SELECT /*+ NO_INDEX */ * FROM " + tableName + " WHERE counter1 >= 0");
```

Comment:
Why not use the index if the index is created?

Reply:
We are already doing one query which asserts that the index is used. This one is to verify that the data table has the same value for the column as the index table.